### DISCUSSION TITLE
谈谈 Rack 的协议与实现 · /rack

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#510] [Announcements]
https://draveness.me/rack

**Comment by hzchirs** [comment]
看過最詳細的 rack 介紹，非常清楚！謝謝分享

**Comment by denglitong** [comment]
最近在看rails的插件化，看到了rails启动过程和engine这块，然后就看到了rack，只能说自己看的太少了，题主讲的很好，很多都讲到了点子上尤其是贴出了源码，向题主学习:)

**Comment by flyfy1** [comment]
照着文章的内容，完整地跟了一遍研究Rack源代码的流程。非常感谢!

一点纠错：

> 其实我们可以发现 Rails 应用就是一堆 Rake 中间件和一个 Rack 应用的集合

应该是

> 其实我们可以发现 Rails 应用就是一堆 *Rack* 中间件和一个 Rack 应用的集合

**Comment by draveness** [comment]
> @flyfy1
> 照着文章的内容，完整地跟了一遍研究Rack源代码的流程。非常感谢!
> 
> 一点纠错：
> 
> > 其实我们可以发现 Rails 应用就是一堆 Rake 中间件和一个 Rack 应用的集合
> 
> 应该是
> 
> > 其实我们可以发现 Rails 应用就是一堆 *Rack* 中间件和一个 Rack 应用的集合

多谢提醒，已经修正了

**Comment by xufeisofly** [comment]
厉害，学习了

**Comment by LongLonghaoran** [comment]
#### 太详细了，分析rackup启动命令开始，翻开源码解释内部启动原理，加载过程，我一直对这块很困惑，读了你文章，视野开阔很多，感谢
